lib/mongo/crypt/auto_decryption_context.rb

```diff
@@ -29,7 +29,7 @@ class AutoDecryptionContext < Context
       #   state machine
       # @param [ String ] db_name The name of the database against which
       #   the command is being made
-      # @param [ Hash ] command The command to be encrypted
+      # @param [ BSON::Document ] command The command to be encrypted
```

Comment:
What this to me is saying is this method expects to be able to access `command` indifferently (i.e. either with symbol or string keys). If so there may need to be a check added that the argument is in fact a Document and not just a Hash. Specifying Hash in the docstring is also legitimate, in which case the method and the caller must agree on whether keys are strings or symbols. I think both approaches are valid, they have different pros/cons.